File: rust/candid_parser/tests/assets/ok/class.rs
// This is an experimental feature to generate Rust binding from Candid.
// You may want to manually adjust some of the types.
#![allow(dead_code, unused_imports)]
use candid::{self, CandidType, Deserialize, Principal};

#[derive(CandidType, Deserialize)]
pub struct List(pub Option<(candid::Int,Box<List>,)>);
#[derive(CandidType, Deserialize)]
pub struct Profile { pub age: u8, pub name: String }

#[ic_cdk::init]
fn init(arg0: candid::Int, l: List, arg2: Profile) {
  unimplemented!()
}
#[ic_cdk::update]
fn get() -> List {
  unimplemented!()
}
#[ic_cdk::update]
fn set(arg0: List) -> List {
  unimplemented!()
}
#[link_section = "icp:public candid:service"]
pub static __SERVICE: [u8; 94] = *br#"type List = opt record { int; List };
service : { get : () -> (List); set : (List) -> (List) }"#;

File: rust/candid_parser/tests/assets/ok/class.js
import { IDL } from '@dfinity/candid';

export const List = IDL.Rec();
List.fill(IDL.Opt(IDL.Tuple(IDL.Int, List)));
export const Profile = IDL.Record({ 'age' : IDL.Nat8, 'name' : IDL.Text });

export const idlService = IDL.Service({
  'get' : IDL.Func([], [List], []),
  'set' : IDL.Func([List], [List], []),
});

export const idlInitArgs = [IDL.Int, List, Profile];

export const idlFactory = ({ IDL }) => {
  const List = IDL.Rec();
  List.fill(IDL.Opt(IDL.Tuple(IDL.Int, List)));
  const Profile = IDL.Record({ 'age' : IDL.Nat8, 'name' : IDL.Text });
  
  return IDL.Service({
    'get' : IDL.Func([], [List], []),
    'set' : IDL.Func([List], [List], []),
  });
};

export const init = ({ IDL }) => {
  const List = IDL.Rec();
  List.fill(IDL.Opt(IDL.Tuple(IDL.Int, List)));
  const Profile = IDL.Record({ 'age' : IDL.Nat8, 'name' : IDL.Text });
  
  return [IDL.Int, List, Profile];
};
